### ISSUE TITLE
Can't figure out image size and number of channels

### ISSUE THREAD

**Issue by anerisheth19** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#50]
When you do the training on COCO dataset, where do you define the image size and how does that map to the convolution size? 

This is in reference to the implementation of MobileNet V2. 
When I change the number of channels to 24 from 32, it throws a runtime error: expected 32 channels, got 24 instead. 

Can you please help me understand the input size and the output after convolutional layers?

**Comment by Daniil-Osokin** [comment]
The input size is defined in `CropPad` augmentation, it is 368x368 by default. Actually there is no relation between convolution and its input spatial size. I don't know the source of error, since you provided no code. If you check the PyTorch [implementation](https://github.com/pytorch/vision/blob/master/torchvision/models/mobilenet.py#L72) of MobileNetV2, it does not have channels number as input parameter, so I suggest not to change it. To use MobileNetV2 as a backbone change [line 92](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/models/with_mobilenet.py#L92) to mobilenet_v2(pretrained=True).features and adjust network stride: change in [line 106](https://github.com/pytorch/vision/blob/master/torchvision/models/mobilenet.py#L106) 2 to 1 to have total stride of 8 (but it's also better to set dilation for succeeding convolution to 2), or just do not use these two last features blocks.

**Comment by anerisheth19** [comment]
Hi, thank you for the clarification. 

I tried this but the accuracy is very low. Working on optimizing it. Thank you for the help.

**Comment by Daniil-Osokin** [comment]
Check, if you have proper input [normalization](https://pytorch.org/docs/stable/torchvision/models.html)  and pass all parameters (`model.parameters()`) to optimizer.

**Comment by anerisheth19** [comment]
I see that you use CropPad augmentation to crop the original image size (eg 640x480) to get to 368x368. Do you think this affects the accuracy? 
In your paper, you mention in Table 2 - Accuracy vs. Complexity of OpenPose on COCO validation set that the max. accuracy is 48.6%. While the original OpenPose has accuracy of around 61%. Is this because of the resolution? Do you know at what resolution is OpenPose getting 61% AP?

**Comment by Daniil-Osokin** [comment]
Did not get the question (in general almost everything affects the accuracy in some way). To train with batches it is necessary to have images with the same size within one batch, thus crop is esential. If you change net input size to smaller one (e.g. from 368 to 64), then network will produce smaller feature maps, and it will be hard to find keypoints there, especially in case of multiple persons. To get this accuracy they used multi-scale testing, then additional refinement with single person pose estimation network.

**Comment by anerisheth19** [comment]
With this CropPad, can we increase the network resolution to say 600x600? Will this improve the validation accuracy is my concern.

**Comment by Daniil-Osokin** [comment]
It might be.

**Comment by anerisheth19** [comment]
> It might be.

So changing the resolution in the part of code below will work right?
```
class CropPad(object):
    def __init__(self, pad, center_perterb_max=40, crop_x=368, crop_y=368):
        self._pad = pad
        self._center_perterb_max = center_perterb_max
        self._crop_x = crop_x
        self._crop_y = crop_y

```

**Comment by Daniil-Osokin** [comment]
Did not do such experiments, you can try it by yourself.
